src/components/Firestore/Requests/types.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CustomThemeType } from '../../../themes';
+import { RulesOutcome } from './rules_evaluation_result_model';
+
+export interface OutcomeData {
```

Comment:
These things are not worth their own interface / type definition since they are only used for the `outcomeData` const. Just let TypeScript infer the type there.